rules/python/lang/weak_hash_dss.yml

```
@@ -0,0 +1,80 @@
+imports:
+  - python_shared_lang_import1
+  - python_shared_lang_datatype
+patterns:
+  - pattern: $<FUNCTION>($<ALGORITHM>, $<OPTIONAL_DATA_TYPE>$<...>)
+    filters:
+      - variable: FUNCTION
+        detection: python_shared_lang_import1
+        scope: cursor
+        filters:
+          - variable: MODULE1
+            values: [hashlib]
+          - variable: NAME
+            values: [new]
+      - variable: ALGORITHM
+        string_regex: (?i)\Adss
```

For my understanding, why don't we match on the end of the string here? 

```suggestion
        string_regex: (?i)\Adss\z
```

---

I think you can use `DSS1` in some cases

---

Ah cool -- thanks 🙏